Revit_Core_Engine/Convert/Environment/ToRevit/Duct.cs

```diff
@@ -0,0 +1,225 @@
+using BH.oM.Spatial.ShapeProfiles;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.MEP.System.SectionProperties;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static Duct ToRevitDuct(this oM.MEP.System.Duct duct, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (document == null)
+                return null;
+
+            // Check valid duct object
+            if (duct == null)
+                return null;
+
+            // Construct Revit Duct
+            Duct revitDuct = refObjects.GetValue<Duct>(document, duct.BHoM_Guid);
+            if (revitDuct != null)
+                return revitDuct;
+
+            // Settings
+            settings = settings.DefaultIfNull();
+
+            // Duct type
+            DuctType ductType = duct.SectionProperty.ToRevitElementType(document, new List<BuiltInCategory> { BuiltInCategory.OST_DuctSystem }, settings, refObjects) as DuctType;
+            if (ductType == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("No valid family has been found in the Revit model. Duct creation requires the presence of the default Duct Family Type.");
+                return null;
+            }
+
+            // End points
+            XYZ start = duct.StartPoint.ToRevit();
+            XYZ end = duct.EndPoint.ToRevit();
+
+            // Level
+            Level level = document.LevelBelow(Math.Min(start.Z, end.Z), settings);
+            if (level == null)
+                return null;
+
+            // Default system used for now
+            // TODO: in the future you could look for the existing connectors and check if any of them overlaps with start/end of this duct - if so, use it in Duct.Create.
+            // hacky/heavy way of getting all connectors in the link below - however, i would rather filter the connecting elements out by type/bounding box first for performance reasons
+            // https://thebuildingcoder.typepad.com/blog/2010/06/retrieve-mep-elements-and-connectors.html
+            MechanicalSystemType mst = new FilteredElementCollector(document).OfClass(typeof(MechanicalSystemType)).OfType<MechanicalSystemType>().FirstOrDefault();
+
+            if(mst == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("No valid MechanicalSystemType can be found in the Revit model. Creating a revit Duct requires a MechanicalSystemType.");
+            }
+
+            BH.Engine.Reflection.Compute.RecordWarning("Duct creation will utilise the first available MechanicalSystemType from the Revit model.");
+
+            SectionProfile sectionProfile = duct.SectionProperty?.SectionProfile;
```

Comment:
you should check if that bad boy is null (the next check, line 90, becomes redundant then)